src/euphorie/client/model.py

```diff
@@ -566,9 +571,10 @@ def allowed(self, object, object_roles=None):
         if object_roles is None:
             return True
 
-        for role in ["Anonymous", "Authenticated", "EuphorieUser"]:
-            if role in object_roles:
-                return True
+        for obj in aq_chain(aq_inner(context)):
+            if IClient.providedBy(obj):
+                allowed_roles = {"Anonymous", "Authenticated", "EuphorieUser", "Reader"}
```

Comment:
Shouldn't this be `allowed_roles = self.getRoles()`?
If we do that, we can also customize `getRoles` for different deployments.
Not sure, if I got this right, though.

Reply:
They are different things.
`getRoles` is there because this class act as a role PAS plugin.
`allowed`  is there because this class act as a PAS member object.

Reply:
As you can see the tests show this is working as expected